cdap-common/src/main/java/io/cdap/cdap/common/lang/InstantiatorFactory.java

```diff
@@ -174,14 +181,28 @@ public T create() {
   }
 
   private <T> Instantiator<T> getByUnsafe(final TypeToken<T> type) {
+    //We do introspection once and then just use reflection to set field values
+    List<Map.Entry<Field, Object>> fieldsToInitialize = new ArrayList<>();
+    Reflections.visit(null, type.getType(), new FieldVisitor() {
+      @Override
```

Comment:
nit: moving this FieldVisitor implementation to a new class

Reply:
Done